vnext/Microsoft.ReactNative/IReactDispatcher.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
+
+  static Mso::DispatchQueue GetUIDispatchQueue(IReactPropertyBag const &properties) noexcept;
+
+  static IReactDispatcher UIThreadDispatcher() noexcept;
```

Comment:
>IReactDispatcher [](start = 9, length = 16)

shouldn't this be a pointer? Or winrt idl is smart for this type of object?

Reply:
C++/WinRT interface-looking types are actually smart pointers to a real ABI interface.

---
In reply to: [423966942](https://github.com/microsoft/react-native-windows/pull/4877#discussion_r423966942) [](ancestors = 423966942)